ontopy/ontology.py

```diff
@@ -1270,6 +1273,60 @@ def addancestors(entity, counter, subject):
             return ancestors.difference(classes)
         return ancestors
 
+    def get_descendants(
+        self,
+        classes: "Union[List, ThingClass]",
+        common: bool = False,
+        generations: int = -1,
+        all_descendants: bool = False,
+    ):
+        """Return descendants/subclasses of all classes in `classes`.
+        Arguments:
+        - classes: to be provided as list.
+        - common: only return common descendants.
+        - generations: Include this number of descendant levels.
+        - all_descendants: Include all descendants.
+        Returns:
+            A list of descendants including required generation.
+            If 'common'=True, the common descendants are returned
+            within the specified number of generations.
+            If 'generations' is not given and 'common' is True all
+            descendants will be checked.
+            'generations' defaults to 1 if 'common' not True.
+        """
+        if (common) & (generations == -1):
+            all_descendants = True
+        elif generations == -1:
+            generations = 1
+
+        if not isinstance(classes, list):
+            classes = [classes]
+
+        descendants = {name: [] for name in classes}
+
+        def _children_recursively(num, newentity, parent, descendants):
+            """Helper function to get all children up to generation."""
+            for child in self.get_children_of(newentity):
+                descendants[parent].append(child)
+                if num < generations:
+                    _children_recursively(num + 1, child, parent, descendants)
+
+        if generations == 0:
+            return set()
+
+        if all_descendants is True:
+            for entity in classes:
+                descendants[entity] = entity.descendants()
+
+        else:
+            for entity in classes:
+                _children_recursively(1, entity, entity, descendants)
+
+        results = [val for _, val in descendants.items()]
```

Comment:
```suggestion
        results = [val for val in descendants.values()]
```

Reply:
It is possible to simplify the logics by skipping the temporary dict by something like the following: 
```python
def get_descendants(self, classes, common=False, generations=None):
    if not isinstance(classes, Sequence):
        classes = [classes]

    def get_children(parent, children, n=0):
        if generations is not None and n < generations:
            for child in self.get_children_of(parent):
                children.add(child)
                get_children(child, children, n+1)
        return children

    children = get_children(classes[0], set())
    for cls in classes[1:]:
        if common:
            children.intersection_update(get_children(cls, set()))
        else:
            children.update(get_children(cls, set()))

    return children
```